driver-core/src/test/java/com/datastax/driver/core/IndexMetadataTest.java

```diff
@@ -0,0 +1,200 @@
+
+import static com.datastax.driver.core.ColumnMetadata.COLUMN_NAME;
+import static com.datastax.driver.core.ColumnMetadata.COMPONENT_INDEX;
+import static com.datastax.driver.core.ColumnMetadata.INDEX_NAME;
+import static com.datastax.driver.core.ColumnMetadata.INDEX_OPTIONS;
+import static com.datastax.driver.core.ColumnMetadata.INDEX_TYPE;
+import static com.datastax.driver.core.ColumnMetadata.KIND;
+import static com.datastax.driver.core.ColumnMetadata.VALIDATOR;
+import static com.datastax.driver.core.DataType.cint;
+import static com.datastax.driver.core.DataType.text;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.datastax.driver.core.ColumnMetadata.IndexMetadata;
+import com.datastax.driver.core.ColumnMetadata.Raw;
+import com.datastax.driver.core.Token.M3PToken;
+import com.datastax.driver.core.utils.CassandraVersion;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Ints;
+
+@CassandraVersion(
+    major = 2.1)
+public class IndexMetadataTest extends CCMBridge.PerClassSingleNodeCluster {
+
+    @Override
+    protected Collection<String> getTableDefinitions() {
+        String createTable = "CREATE TABLE indexing ("
+            + "id int primary key,"
+            + "map_values map<text, int>,"
+            + "map_keys map<text, int>,"
+            + "map_entries map<text, int>,"
+            + "map_full frozen<map<text, int>>,"
+            + "set_full frozen<set<text>>,"
+            + "list_full frozen<list<text>>,"
+            + "custom text)";
+        return ImmutableList.of(createTable);
+    }
+
+    @Test(
+        groups = "short")
+    public void should_not_flag_map_index_type() {
+        String createValuesIndex = String.format("CREATE INDEX map_values_index ON %s.indexing (map_values);", keyspace);
+        session.execute(createValuesIndex);
+        IndexMetadata index = getIndexForColumn("map_values");
+        Assert.assertEquals(index.getName(), "map_values_index");
+        Assert.assertFalse(index.isKeys());
+        Assert.assertFalse(index.isFull());
+        Assert.assertFalse(index.isEntries());
+        Assert.assertFalse(index.isCustomIndex());
+        Assert.assertEquals(index.asCQLQuery(), createValuesIndex);
+    }
+
+    @Test(
+        groups = "short")
+    public void should_flag_map_index_type_as_keys() {
+        String createKeysIndex = String.format("CREATE INDEX map_keys_index ON %s.indexing (KEYS(map_keys));", keyspace);
+        session.execute(createKeysIndex);
+        IndexMetadata index = getIndexForColumn("map_keys");
+        Assert.assertEquals(index.getName(), "map_keys_index");
+        Assert.assertTrue(index.isKeys());
+        Assert.assertFalse(index.isFull());
+        Assert.assertFalse(index.isEntries());
+        Assert.assertFalse(index.isCustomIndex());
+        Assert.assertEquals(index.asCQLQuery(), createKeysIndex);
```

Comment:
We're progressively migrating to AssertJ, could you use it?

```java
import static com.datastax.driver.core.Assertions.assertThat;
...
assertThat(index.getName()).isEqualTo("map_keys_index");
```

We could even have a custom assertion class for `IndexMetadata` (see existing examples like `DataTypeAssert`), although if you're short on time I'm fine with testing the properties individually.